backend/uclapi/workspaces/tests_archive.py

```diff
@@ -0,0 +1,270 @@
+        self.assertEqual(len(survey_change), 2)
+        sensor_replacement = survey_change[1]
+
+        del sensor_replacement["id"]
+        del sensor_replacement["datetime"]
+        self.assertDictEqual(sensor_replacement, {"active": True,
+                                                  "end_datetime": datetime(2050, 1, 1, 15, 0),
+                                                  "name": "Replacement Room 2",
+                                                  "start_datetime": datetime(2010, 1, 1, 9, 0),
+                                                  "survey_id": 10})
+
+    def tearDown(self):
+        Surveys.objects.all().delete()
+        Sensors.objects.all().delete()
+        Historical.objects.all().delete()
+        SensorReplacements.objects.all().delete()
+        SurveyChanges.objects.all().delete()
+
+
+@freeze_time("2020-02-01")
+class OccupEyeArchiveViewsTestCase(TestCase):
+
+    @patch("workspaces.occupeye.archive.FIRST_OCCUPEYE_INSTALLATION",
+           datetime.strptime("2020-01-01T00:00:00+0000", "%Y-%m-%dT%H:%M:%S%z"))
+    @patch("workspaces.occupeye.archive.MAX_TIME_DELTA", timedelta(days=7))
+    def setUp(self):
+        with open(os.path.join(__location__, "tests_archive_cache.json"), encoding="utf-8") as f:
+            self.archive = OccupEyeArchive(endpoint=TestEndpoint(json.load(f)))
+        self.archive.reset()
+        self.archive.update()
+        self.api = OccupEyeApi()
+        self.client = Client()
+
+        try:
+            user = User.objects.get(email="develop@ucl.ac.uk")
+        except User.DoesNotExist:
+            user = User(
+                email="develop@ucl.ac.uk",
+                full_name="UCL API Developer",
+                given_name="UCL API",
+                department="Dept of API Development",
+                cn="develop",
+                raw_intranet_groups="ucl-all;ucl-ug;schsci-all",
+                employee_id="uclapi1"
+            )
+            user.save()
+
+        try:
+            app = App.objects.get(user=user, name="Local OAuth Test")
+        except App.DoesNotExist:
+            app = App(
+                user=user,
+                name="Local OAuth Test",
+                api_token="uclapi-4286bc18b235d86-ab0998cc3a47a9b-07b6dfe234a04bf-97407a655b33ae8",
+                client_id="1105308584328350.9460393713696551",
+                client_secret="251e9f9553bb3b86829c18bf795844d977dedf569b24a70e4d4e753958fcc2f3",
+                callback_url="http://localhost:8002/uclapi/callback"
+            )
+            app.save()
+
+    def test_get_historical_sensor(self):
+        start_time = datetime.strptime("2020-01-15T13:00", "%Y-%m-%dT%H:%M")
+        end_time = datetime.strptime("2020-01-15T14:00", "%Y-%m-%dT%H:%M")
+
+        response = self.client.get("/workspaces/historical/data",
+                                   {"token": "uclapi-4286bc18b235d86-ab0998cc3a47a9b-07b6dfe234a04bf-97407a655b33ae8",
```

Comment:
```suggestion
                                   {"token": self.app.api_token,
```
Ditto for other tests.